lib/logging.h

```diff
@@ -38,7 +38,7 @@ void qlog(int priority, const char *format, ...) __PRINTF__(2, 3);
 	} while(0)
 
 #ifdef __cplusplus
-extern "C" {
```

Comment:
This would indicate that patch 2 doesn't build. Can you please update that instead?